Merge #1814 after JIT changes

bettio opened the pull request:
PR #1814 fails after JIT changes, apply fix before merging it to main.
Also some other improvements have been done.

Closes #1814

These changes are made under both the "Apache 2.0" and the "GNU Lesser General
Public License 2.1 or later" license terms (dual license).

SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later

bettio commented on tests/test.c:
```diff
@@ -544,6 +544,9 @@ struct Test tests[] = {
 
     TEST_CASE(test_code_all_available_loaded),
     TEST_CASE_EXPECTED(test_code_load_binary, 24),
+#ifndef AVM_NO_JIT
```
@pguyot This is my best try at fixing build error with [build-and-test (macos-15-intel, 28, -DAVM_DISABLE_JIT=OFF)](https://github.com/atomvm/AtomVM/actions/runs/18185429333/job/51793575084?pr=1814#logs). Can we do anything better than just skipping the test?

pguyot replied:
This is the bug: https://github.com/atomvm/AtomVM/pull/1814/files#r2399909494

Additionally, maybe JIT is not very tolerant to hot code reloading on macOS (the mmap didn't have the proper flags), but it's not like we support hot code reloading yet.

FranekKubis committed:
Implement code:get_object_code/1

This function works with a caveat, unlike on the BEAM, it returns a
valid result even for modules loaded with `code:load_binary/3`.
Fixing this right now, would mean making bigger changes.

Co-authored-by: Davide Bettio <davide@uninstall.it>
Signed-off-by: Davide Bettio <davide@uninstall.it>

pguyot commented on src/libAtomVM/module.c:
```diff
@@ -1114,6 +1114,8 @@ Module *module_new_from_iff_binary(GlobalContext *global, const void *iff_binary
     mod->fun_table = beam_file + offsets[FUNT];
     mod->str_table = beam_file + offsets[STRT];
     mod->str_table_len = sizes[STRT];
+    mod->binary = beam_file;
```
We could conditionally set this if module is loaded from file or from a binary and thus solve the (main) incompatibility with BEAM

bettio replied:
That is an option, but either means changing all `module_new_from_iff_binary` calls or manually adding that change to different places in all platforms, that sounds messy.

Actually we need a refactor, that separates: finding a beam file in .avm or filesystem, from loading it and from parsing it and creating a new Module struct. That is something I wouldn't do right now.

A proper code:get_object_code would require implementing find + load to binary, not parsing the module and everything else.
